opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -112,6 +115,133 @@ def force_flush(self, timeout_millis: int = 30000):
         """
 
 
+class SynchronousMultiLogProcessor(LogProcessor):
+    """Implementation of class:`LogProcessor` that forwards all received
+    events to a list of log processors sequentially.
+
+    The underlying log processors are called in sequential order as they were
+    added.
+    """
+
+    def __init__(self):
+        # use a tuple to avoid race conditions when adding a new log and
+        # iterating through it on "emit".
+        self._log_processors = ()  # type: Tuple[LogProcessor, ...]
+        self._lock = threading.Lock()
+
+    def add_log_processor(self, log_processor: LogProcessor) -> None:
+        """Adds a Logprocessor to the list of log processors handled by this instance"""
+        with self._lock:
+            self._log_processors = self._log_processors + (log_processor,)
+
+    def emit(self, log_data: LogData) -> None:
+        for lp in self._log_processors:
```

Comment:
Is a lock also necessary here?

Reply:
No, we wouldn't want to hold the lock until all processors complete emit. This just iterates over the log processor available at the time of emitting and returns.